File: training task/4_hierachical_inheritance.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def get_details(self):
        return f"Name: {self.name}, Age: {self.age}"

class Student(Person):
    def __init__(self, name, age, student_id):
        super().__init__(name, age)
        self.student_id = student_id

    def get_student_details(self):
        person_details = self.get_details()
        return f"{person_details}, Student ID: {self.student_id}"

class Teacher(Person):
    def __init__(self, name, age, employee_id):
        super().__init__(name, age)
        self.employee_id = employee_id

    def get_teacher_details(self):
        person_details = self.get_details()
        return f"{person_details}, Employee ID: {self.employee_id}"

student = Student("Nitu", 20, "S19")
teacher = Teacher("sadia", 35, "T89")

print(student.get_student_details())
print(teacher.get_teacher_details())

File: training task/4_hierachical_inheritance.cpp
#include <iostream>
#include <string>

using namespace std;
class Person {
protected:
    string name;
    int age;

public:
    Person(string name, int age) {
        this->name = name;
        this->age = age;
    }
    string getDetails() const {
        return "Name: " + name + ", Age: " + to_string(age);
    }
};


class Student : public Person {
private:
    string student_id;

public:
    Student(string name, int age, string student_id)
        : Person(name, age) {
            this->student_id=student_id;
        }

   
    string getStudentDetails() const {
        string person_details = getDetails();
        return person_details + ", Student ID: " + student_id;
    }
};

class Teacher : public Person {
private:
    string employee_id;

public:
    Teacher(string name, int age, string employee_id)
        : Person(name, age) {
            this->employee_id=employee_id;
        }


    string getTeacherDetails() const {
        string person_details = getDetails();
        return person_details + ", Employee ID: " + employee_id;
    }
};

int main() {
  
    Student student("Nitu", 20, "S19");
    Teacher teacher("sadia", 35, "T89");

    cout << student.getStudentDetails() << endl;
    cout << teacher.getTeacherDetails() << endl;

    return 0;
}
